RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterQuickFixHelper.cs

```diff
@@ -0,0 +1,195 @@
+
+            int stopIndex = startIndex + toReplace.Length - 1;
+            var prefix = original.Substring(0, startIndex);
+            var suffix = (stopIndex >= original.Length) ? string.Empty : original.Substring(stopIndex + 1);
+            var toBeReplaced = original.Substring(startIndex, stopIndex - startIndex + 1);
+
+            Debug.Assert(toBeReplaced.IndexOf(toReplace) == 0);
+            return prefix + toBeReplaced.Replace(toReplace, replacement) + suffix;
+        }
+        public bool VariableNameIsAlreadyUsed(string variableName)
+        {
+            if(_namesInUse.Count() == 0)
+            {
+                _namesInUse = GetVariableNamesUsedInProcedureContext();
+            }
+            return _namesInUse.Any(name => variableName.ToUpper().Equals(name.ToUpper()));
+        }
+
+        public IReadOnlyList<VBAParser.BlockStmtContext> GetBlockStmtContextsForContext(RuleContext context)
+        {
+            if (context is VBAParser.SubStmtContext)
+            {
+                return ((VBAParser.SubStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.FunctionStmtContext)
+            {
+                return ((VBAParser.FunctionStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertyLetStmtContext)
+            {
+                return ((VBAParser.PropertyLetStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertyGetStmtContext)
+            {
+                return ((VBAParser.PropertyGetStmtContext)context).block().blockStmt();
+            }
+            else if (context is VBAParser.PropertySetStmtContext)
+            {
+                return ((VBAParser.PropertySetStmtContext)context).block().blockStmt();
+            }
+            return null;
+        }
+
+        public IReadOnlyList<VBAParser.ArgContext> GetArgContextsForContext(RuleContext context)
+        {
+            if (context is VBAParser.SubStmtContext)
+            {
+                return ((VBAParser.SubStmtContext)context).argList().arg();
+            }
+            else if (context is VBAParser.FunctionStmtContext)
+            {
+                return ((VBAParser.FunctionStmtContext)context).argList().arg();
+            }
+            else if (context is VBAParser.PropertyLetStmtContext)
+            {
+                return ((VBAParser.PropertyLetStmtContext)context).argList().arg();
+            }
+            else if (context is VBAParser.PropertyGetStmtContext)
+            {
+                return ((VBAParser.PropertyGetStmtContext)context).argList().arg();
+            }
+            else if (context is VBAParser.PropertySetStmtContext)
+            {
+                return ((VBAParser.PropertySetStmtContext)context).argList().arg();
+            }
+            return null;
+        }
+        private HashSet<string> GetVariableNamesUsedInProcedureContext()
```

Comment:
again: type seems overspecified. 
Also: Why not access the parser-state here and query the Declarations?